src/table/utils/styling-utils.ts

```diff
@@ -81,7 +89,7 @@ export const getBaseStyling = (
     fontFamily,
     color: isColorSet(styleObj?.fontColor) ? getColor(STYLING_DEFAULTS.FONT_COLOR, theme, styleObj?.fontColor) : color,
     fontSize: (styleObj?.fontSize && `${styleObj.fontSize}px`) || fontSize,
-    padding: getPadding(styleObj),
+    padding: styleObj && 'padding' in styleObj ? getPadding(styleObj) : undefined,
```

Comment:
once we add padding to the API, I think it should be a string. Therefor we could use optional chainging here
```suggestion
    padding: styleObj?.padding ? getPadding(styleObj) : undefined,
```

Reply:
Unfortuanlly typescript complains `styleObj?.padding`

Reply:
hmm ok I have no better suggestion then

Reply:
I have a question. When we set a fontSize for the header, the header should get padding based on that font size, right?

Reply:
ah yes, we need to run it even if there is no padding. Now you are basically doing the logic that is in the getPadding function before you call it. I would say better to rewrite getPadding to do better sanity checking

Reply:
sounds good. Doing it

Reply:
Done